src/Analytics/UAExperimentData.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace PhpAb\Analytics;
+
+use Webmozart\Assert\Assert;
+
+class UAExperimentData
+{
+
+    /**
+     * @var array Of test identifiers and their variant's indexes
+     */
+    private $testsData = [];
+
+    /**
+     * @param string $testIdentifier It will look like "Qp0gahJ3RAO3DJ18b0XoUQ"
+     * @param int $variationIndex
+     * @return \PhpAb\Analytics\UAExperiments
+     * @throws InvalidArgumentException
+     */
+    public function onRegisterParticipation($testIdentifier, $variationIndex)
+    {
+        Assert::string($testIdentifier, 'Test identifier must be astring');
+
+        Assert::integer($variationIndex, 'Variation index must be integer');
```

Comment:
It will be a concern for when we link all actors together.
In any case, this class shouldn't know about anything else than the data it's injected into, in this case the variation index.